mmv1/products/cloudrun/api.yaml

```diff
@@ -616,6 +616,68 @@ objects:
                     required: true
                     description: |-
                       This must match the Name of a Volume.
+            - !ruby/object:Api::Type::NestedObject
+              name: startupProbe
+              min_version: beta
+              description: |-
+                Startup probe of application within the container.
+                All other probes are disabled if a startup probe is provided, until it
+                succeeds. Container will not be added to service endpoints if the probe fails.
+              properties:
+                - !ruby/object:Api::Type::Integer
+                  name: initialDelaySeconds 
+                  description: |-
+                    Number of seconds after the container has started before the probe is
+                    initiated.
+                    Defaults to 0 seconds. Minimum value is 0. Maximum value is 240.
+                - !ruby/object:Api::Type::Integer
+                  name: timeoutSeconds 
+                  description: |-
+                    Number of seconds after which the probe times out.
+                    Defaults to 1 second. Minimum value is 1. Maximum value is 3600.
+                    Must be smaller than periodSeconds.
+                - !ruby/object:Api::Type::Integer
+                  name: periodSeconds 
+                  description: |-
+                    How often (in seconds) to perform the probe.
+                    Default to 10 seconds. Minimum value is 1. Maximum value is 240.
+                - !ruby/object:Api::Type::Integer
+                  name: failureThreshold  
+                  description: |-
+                    Minimum consecutive failures for the probe to be considered failed after
+                    having succeeded. Defaults to 3. Minimum value is 1.
+                - !ruby/object:Api::Type::NestedObject
+                  name: tcpSocket 
+                  description: |-
+                    TcpSocket speficies an action involving a TCP port.
+                  properties:
+                    - !ruby/object:Api::Type::Integer
+                      name: port
+                      description: |-
+                        Port number to access on the container. Number must be in the range 1 to 65535.
+                - !ruby/object:Api::Type::NestedObject
+                  name: httpGet 
+                  description: |-
+                    HttpGet specifies the http request to perform..
+                  properties:
+                    - !ruby/object:Api::Type::String
+                      name: path
```

Comment:
Can this value ever be empty? It seems likely that it should be required.

Reply:
Added a test to cover it.

Reply:
We don't appear to be updating the field in the test- see the comment about `default_from_api` on the field, we likely need a `default_value` instead.

I think Terraform Core's eating the diff there- it's considering `""` equal to unset and preserving the old value.

Reply:
I don't think it's considering "" equal to unset. I tried `default_value = "/"`. The test with `path = ""` failed.

Reply:
Ah! I see what's happening- we're sending the following in the update:

```
"startupProbe": {
       "failureThreshold": 3,
       "httpGet": {
        "httpHeaders": [
         {
          "name": "User-Agent",
          "value": "magic-modules"
         }
        ]
       },
       "periodSeconds": 10,
       "timeoutSeconds": 1
      }
```

Notice that `path` isn't present. While this is a PUT, and per normal PUT rules we'd expect lack of presence to mean the type's empty value, this isn't interpreted that way by the API. It's either preserving the value or setting the default value instead, likely preserving it. That means that we see the following on return:

```
            "startupProbe": {
              "timeoutSeconds": 1,
              "periodSeconds": 10,
              "failureThreshold": 3,
              "httpGet": {
                "path": "/",
                "httpHeaders": [
                  {
                    "name": "User-Agent",
                    "value": "magic-modules"
                  }
                ]
              }
```

We can resolve that by setting `send_empty_value: true` on the field, as described at https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1#apiyaml